packages/itwin/grouping-mapping-widget/src/components/Mappings/Extraction/ExtractionMessageModal.tsx

```diff
@@ -20,7 +24,55 @@ export interface ExtractionMessageModalProps {
 }
 
 export const ExtractionMessageModal = ({ isOpen, onClose, extractionMessageData, timestamp }: ExtractionMessageModalProps) => {
+  const [formattedExtractionMessage, setFormattedExtractionMessage] = useState<ExtractionMessageData[]>([]);
+  const groupingMappingApiConfig = useGroupingMappingApiConfig();
+  const mappingClient = useMappingClient();
+  const { mappings } = useMappingsOperations({...groupingMappingApiConfig, mappingClient});
   const [formattedTimestamp, setFormattedTimestamp] = useState<string>("");
+
+  const getMappingName = useCallback(async (mappingId: string) => {
+    return mappings.find((mapping) => {return mapping.id === mappingId;})?.mappingName ?? "";
+  }, [mappings]);
+
+  const getGroupNames = useCallback(async (mappingId: string, groupId: string, groupsCache: Map<string, Group[]>) => {
+    const accessToken = await groupingMappingApiConfig.getAccessToken();
+    if(!groupsCache.has(mappingId)){
+      const groups = await mappingClient.getGroups(
+        accessToken,
+        groupingMappingApiConfig.iModelId,
+        mappingId
+      );
+      groupsCache.set(mappingId, groups);
+    }
+    return groupsCache.get(mappingId)?.find((group) => {return group.id === groupId;})?.groupName ?? "";
+  }, [groupingMappingApiConfig, mappingClient]);
+
+  useEffect(() => {
+    const formatMessages = async () => {
+      const groupsCache = new Map<string,Group[]>();
+      const extractionMessageDataPromises = extractionMessageData.map(async (extractionMessage) => {
+        let replacedMessage = extractionMessage.message;
+        if(extractionMessage.message.includes("iModel")){
+          replacedMessage = replacedMessage.replace(/iModel [\w-]+/, "iModel");
+        }
+        if(replacedMessage.includes("MappingId:")){
+          const splittedMessage = replacedMessage.split(" ");
+          const mappingId = splittedMessage[splittedMessage.indexOf("MappingId:") + 1];
+          const groupId = splittedMessage[splittedMessage.indexOf("GroupId:") + 1];
+          const mappingName = await getMappingName(mappingId);
```

Comment:
`getMappingNmae` doesn't need to be awaited.

Reply:
I have had it removed.